### ISSUE TITLE
ProviderID should be provided to successfully initialise nodes

### ISSUE THREAD

**Issue by irozzo-1A** [kubernetes-sigs/cloud-provider-huaweicloud#74]
Hi,

I tried to use this CCM in a test cluster and I observed that the nodes did not get initialised i.e. taint `node.cloudprovider.kubernetes.io/uninitialized` was never removed. This is the root issue:

```
E0629 09:47:05.321556       1 node_controller.go:364] failed to set node provider id: failed to get instance ID from cloud provider: unimplemented
```

In fact checking the code I noticed that method [InstanceID](https://github.com/kubernetes-sigs/cloud-provider-huaweicloud/blob/f16d76d/pkg/cloudprovider/huaweicloud/instances.go#L103) is not implemented (among others). After manually patching the node to add the `ProviderID` it was initialised successfully.

I think that this is necessary to be able to use this CCM as either manually patching the nodes or setting explicitly the kubelet flag `--provider-id` don't seem to be viable options.

Additional info:
Kubernetes version: `v1.17.5`

**Comment by RainbowMango** [comment]
Hi @irozzo-1A,
Can you share more logs and `kubelet`'s `--provider-id` parameter?

**Comment by irozzo-1A** [comment]
> Hi @irozzo-1A,
> Can you share more logs and `kubelet`'s `--provider-id` parameter?

Hi @RainbowMango, thx for replying.
In my environment I did not set kubelet `--provider-id` flag. I was not expecting it to be mandatory as with other CCMs I used it is not, because `InstanceID` method is implemented (e.g. [OpenStack CCM](https://github.com/kubernetes/cloud-provider-openstack/blob/master/pkg/cloudprovider/providers/openstack/openstack_instances.go#L174)).

Should I conclude that kubelet flag`--provider-id` is mandatory with this CCM? I did not find anything mentioning it, but to be fair in the local example [it is specified](https://github.com/kubernetes-sigs/cloud-provider-huaweicloud/blob/master/docs/quick-start-with-local-cluster.md).

If you agree that it would be more handy not to leave the burden of specifying the provider ID to the user I could submit a PR to implement `InstanceID` method. WDYT?
